src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,419 @@
+     */
+    @ApiStatus.Internal
+    public static boolean isOpenBraceCharacter(@NotNull PsiFile file, @Nullable Character character) {
+        return (character != null) && getBracePairsFwd(file).containsKey(character);
+    }
+
+    /**
+     * Determines whether or not the specified character can close a brace pair in the provided file.
+     *
+     * @param file      the PSI file
+     * @param character the optional character
+     * @return true if the character can close a brace pair in the file; otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isCloseBraceCharacter(@NotNull PsiFile file, @Nullable Character character) {
+        return (character != null) && getBracePairsBwd(file).containsKey(character);
+    }
+
+    /**
+     * Returns the brace pair open character for the specified brace pair close character in the provided file.
+     *
+     * @param file                the PSI file
+     * @param closeBraceCharacter the optional brace pair close character
+     * @return the corresponding brace pair open character if found; otherwise null
+     */
+    @Nullable
+    @ApiStatus.Internal
+    public static Character getOpenBraceCharacter(@NotNull PsiFile file, @Nullable Character closeBraceCharacter) {
+        return closeBraceCharacter != null ? getBracePairsBwd(file).get(closeBraceCharacter) : null;
+    }
+
+    /**
+     * Returns the brace pair close character for the specified brace pair open character in the provided file.
+     *
+     * @param file               the PSI file
+     * @param openBraceCharacter the optional brace pair open character
+     * @return the corresponding brace pair close character if found; otherwise null
+     */
+    @Nullable
+    @ApiStatus.Internal
+    public static Character getCloseBraceCharacter(@NotNull PsiFile file, @Nullable Character openBraceCharacter) {
+        return openBraceCharacter != null ? getBracePairsFwd(file).get(openBraceCharacter) : null;
+    }
+
+    @NotNull
+    private static Map<Character, Character> getBracePairsFwd(@NotNull PsiFile file) {
+        return CachedValuesManager.getCachedValue(file, () -> CachedValueProvider.Result.create(getBracePairs(file), file));
+    }
+
+    @NotNull
+    private static Map<Character, Character> getBracePairsBwd(@NotNull PsiFile file) {
+        return CachedValuesManager.getCachedValue(file, () -> CachedValueProvider.Result.create(ContainerUtil.reverseMap(getBracePairsFwd(file)), file));
+    }
+
+    // Comments
+
+    @NotNull
+    public static Commenter getCommenter(@NotNull PsiFile file) {
+        Commenter commenter = null;
+
+        if (isAbstractFileTypeFile(file)) {
+            AbstractFileType abstractFileType = (AbstractFileType) file.getFileType();
+            commenter = abstractFileType.getCommenter();
+        }
+
+        // NOTE: Unfortunately the language grammar comment information is not available for TextMate bundles
```

Comment:
As stated here, unfortunately TextMate bundles don't provide any first-class notion of the comment prefix/suffix information.